crates/red_knot_python_semantic/src/types.rs

```diff
@@ -872,6 +890,7 @@ impl<'db> Type<'db> {
             Type::StringLiteral(str) => Truthiness::from(!str.value(db).is_empty()),
             Type::LiteralString => Truthiness::Ambiguous,
             Type::BytesLiteral(bytes) => Truthiness::from(!bytes.value(db).is_empty()),
+            Type::SliceLiteral(_) => Truthiness::AlwaysTrue,
```

Comment:
`slice` appears to have no custom `__bool__` logic; even empty slices and things like `slice(None, None)` are `True`.